lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -172,19 +188,29 @@ data StakePoolLayer = StakePoolLayer
         -- Exclude all pools that retired in or before this epoch.
         -> Coin
         -> ExceptT ErrListPools IO [Api.ApiStakePool]
+
+    , putSettings :: Settings -> IO ()
+
+    , getSettings :: IO Settings
     }
 
 newStakePoolLayer
     :: forall sc. ()
     => NetworkLayer IO (IO Shelley) (CardanoBlock sc)
     -> DBLayer IO
-    -> StakePoolLayer
-newStakePoolLayer nl db@DBLayer {..} =
-        StakePoolLayer
-            { getPoolLifeCycleStatus = _getPoolLifeCycleStatus
-            , knownPools = _knownPools
-            , listStakePools = _listPools
-            }
+    -> (MVar Bool -> IO (Async ()))
+    -> IO StakePoolLayer
+newStakePoolLayer nl db@DBLayer {..} metadataRun' = do
+    mv <- newMVar False
+    as <- metadataRun' mv
+    mas <- newMVar as
```

Comment:
Those variables are used across multiple functions and the name are sort of cryptic :thinking:
In particular, it is unclear why the second mvar `mas` is needed?